pkg/controller/machine/machine_controller.go

```diff
@@ -116,6 +116,7 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 	// for machine management.
 	cluster, err := r.getCluster(ctx, m)
 	if err != nil {
+		klog.Infof("No active cluster found for machine %q/%q", m.Namespace, m.Name)
```

Comment:
`getCluster` does emit a log message if the cluster label is missing. The only way it returns an error is if the call to `r.client.Get` failed. I'm thinking it makes more sense to move this log message into `getCluster`, and also include the value of the label - something like this:
```go
klog.Errorf("Error getting cluster %q for machine %q in namespace %q: %v", machine.Labels[clusterv1.MachineClusterLabelName], machine.Name, machine.Namespace)
```

Or it might be better to return a wrapped error (`errors.Wrapf()`) with this information and let the error bubble up to the controller, which will log it.

WDYT?